src/server/game/AI/CoreAI/UnitAI.h

```diff
@@ -148,15 +148,21 @@ class TC_GAME_API UnitAI
         {
             ThreatContainer::StorageType const& threatlist = me->getThreatManager().getThreatList();
             if (position >= threatlist.size())
-                return NULL;
+                return nullptr;
 
             std::list<Unit*> targetList;
+            HostileReference* currentVictim = me->getThreatManager().getCurrentVictim();
+
+            // Current victim always goes first
+            if (predicate(currentVictim))
```

Comment:
you want to use `.getCurrentVictim()->getTarget()` to deal with units instead of HostileReference